src/engine/BMGame.php

```diff
@@ -939,8 +821,10 @@ protected function roll_active_dice_needing_values() {
 
     protected function update_game_state_specify_dice() {
         if (!$this->isWaitingOnAnyAction()) {
-            $this->prevSwingValueArrayArray = NULL;
-            $this->prevOptValueArrayArray = NULL;
+            foreach ($this->playerArray as $player) {
+                $player->prevSwingValueArray = array();
+                $player->prevOptValueArray = array();
+            }
```

Comment:
Huh, okay, i had somehow gotten it into my head that the issue was that foreach can't iterate over a zero-length array.  If the issue is just that foreach can't treat NULL as an array, that's substantially less terrible of PHP.  :>)  Anyway, makes sense.